### PR TITLE
[WIP] Switch versions of go-jose and go-oidc

### PR DESCRIPTION
**Pull request by dims**
- `square/go-jose` has been `archived by the owner` since Feb 27, 2023 there is an active community fork here now https://github.com/go-jose/go-jose/
- `coreos/go-oidc` switched over to `go-jose/go-jose` on Dec 27, 2022 in https://github.com/coreos/go-oidc/pull/360

We are late to the party :)

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by enj** [comment]
/hold

There are behavioral changes in go-oidc v3 that don't match what I would expect from KAS (will link the old PR later).

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/123253#issuecomment-1940379993

https://github.com/kubernetes/kubernetes/pull/117437#issuecomment-1520735668

https://github.com/kubernetes/kubernetes/pull/114772#pullrequestreview-1515045221

Not opposed to making the switch but it needs super careful review which we haven't had bandwidth for yet

File: test/integration/auth/svcaccttoken_test.go [hunk 1]
```diff
@@ -228,7 +228,7 @@ func TestServiceAccountTokenCreate(t *testing.T) {
 		}
 
 		checkPayload(t, treq.Status.Token, `"system:serviceaccount:myns:test-svcacct"`, "sub")
-		checkPayload(t, treq.Status.Token, `["api"]`, "aud")
+		checkPayload(t, treq.Status.Token, `"api"`, "aud")
```
Comment (enj) [inline test/integration/auth/svcaccttoken_test.go:231]: No payload should change on a dep bump.

**Comment by dims** [comment]
apologies, meant to put WIP, but forgot. definitely NOT for 1.30

**Pull request closed without merging**